components/pedigree/resources/src/main/resources/pedigree/controller.js

```diff
@@ -149,6 +150,85 @@ define([
+            var callbackOnApprove = event.memo.callbackOnApprove;
+            var phenotipsIdListToRemove = event.memo.patientProfilesToBeRemoved;
+
+            var composePatientList = function(list, convertIdToLink) {
+                var result = null;
+                list.forEach(function(id){
+                    result = (result == null ? "" : (result + ", "));
+                    result += convertIdToLink ? editor.getExternalEndpoint().getPhenotipsPatientHTMLLink(id) : id;
+                });
+                return result;
+            }
+
+            var patientListString = composePatientList(phenotipsIdListToRemove, true);
+
+            var title       = "Delete patient record" + (phenotipsIdListToRemove.length > 1 ? "s" : "") + "?";
+            var buttonLabel = "Delete" + (phenotipsIdListToRemove.length > 1 ? " all listed records" : "");
+            var message     = "The deletion of " + (phenotipsIdListToRemove.length > 1 ? "patient records" : "a patient record") +
+                              " (" + patientListString + ") is not reversible. Are you sure you wish to continue?";
+
+            var deleteRecords = function() {
+
+                // perform the callback regardless of if deletion was successful
+                callbackOnApprove();
+
+                console.log("Deleting patient record(s) " + phenotipsIdListToRemove);
+
+                var failedDeletions = [];
+                var successfulDeletions = [];
+
+                var reportErrorsIfAny = function() {
+                    if (failedDeletions.length > 0) {
+                        if (phenotipsIdListToRemove.length == 1) {
+                            var id = phenotipsIdListToRemove[0];
+                            var title = "Error deleting patient record " + id;
+                            var message = "Failed to delete patient record " + id + ". You might not have permission to do so.";
+                        } else {
+                            var title = "Could not delete some of the patient records";
+                            var message = "Failed to delete the following patient records: " + composePatientList(failedDeletions, true)
+                            + ".<br><br>(note: the following patient records were successfully removed: "
+                            + composePatientList(successfulDeletions) + ")";
+                        }
+                        editor.getOkCancelDialogue().showError(message, title, "OK", null);
+                    }
+                }
+
+                var completed = 0;
+                phenotipsIdListToRemove.forEach(function(phenotipsPatientID) {
+                    new Ajax.Request(editor.getExternalEndpoint().getPatientDeleteURL(phenotipsPatientID), {
+                        method: 'POST',
+                        onCreate: function() {
+                            document.fire("pedigree:load:start", {"message": "Deleting patient " + phenotipsPatientID + "..."});
```

Comment:
Sending these events like this won't work nicely. Since the loading modal is a singleton, if 3 patients are deleted, this event will be sent 3 times in a row, but only one dialog is shown, with only the last patient ID shown. Then the first deletion to complete will close the dialog, even though other requests are in progress. Maybe trigger the dialog once before the loop, and close it when all requests are completed.

Reply:
The event should definitely be renamed, I just did nto do it as part of this PR, but since I modified relevant code I probably should have

I remember I had some reasoning why a few events in a row is not a problem, but let me re-think